tests/unit/fold_matrix_to_scalar.cpp

```diff
@@ -0,0 +1,657 @@
+		std::cout << "-- Running test 03: Square matrix of size n = " << n << ", with n 1s on the first column" << std::endl;
+		input< NzType, void > input(I, mask);
+		if( launcher.exec( &grb_program, input, rc, true ) != SUCCESS ) {
+			std::cerr << "Launching test 03 FAILED\n";
+			return 255;
+		}
+		std::cout << std::endl << std::flush;
+	}
+
+	if( ! rc ) { // Building a square-matrix with n 1s on the first row and column
+		Matrix< NzType > I( n, n );
+		std::vector< size_t > I_rows( 2 * n - 1, 0 ), I_cols( 2 * n - 1, 0 );
+		std::vector< NzType > I_vals( 2 * n - 1, 1.f );
+		std::iota( I_rows.begin() + n, I_rows.end(), 1 );
+		std::iota( I_cols.begin(), I_cols.begin() + n, 0 );
+		buildMatrixUnique( I, I_rows.data(), I_cols.data(), I_vals.data(), I_vals.size(), SEQUENTIAL );
+		Matrix< void > mask( n, n );
+		buildMatrixUnique( mask, I_rows.data(), I_cols.data(), I_rows.size(), SEQUENTIAL );
+		std::cout << "-- Running test 04: Square matrix of size n = " << n << ", with n 1s on the first row and column" << std::endl;
+		input< NzType, void > input(I, mask);
+		if( launcher.exec( &grb_program, input, rc, true ) != SUCCESS ) {
+			std::cerr << "Launching test 04 FAILED\n";
+			return 255;
+		}
+		std::cout << std::endl << std::flush;
+	}
+
+	if( ! rc ) { // Building a [1 row, n columns] matrix filled with 1s
+		Matrix< NzType > I( 1, n );
+		std::vector< size_t > I_rows( n, 0 ), I_cols( n, 0 );
+		std::vector< NzType > I_vals( n, 1.f );
+		std::iota( I_cols.begin(), I_cols.end(), 0 );
+		buildMatrixUnique( I, I_rows.data(), I_cols.data(), I_vals.data(), I_vals.size(), SEQUENTIAL );
+		Matrix< void > mask( 1, n );
+		buildMatrixUnique( mask, I_rows.data(), I_cols.data(), I_rows.size(), SEQUENTIAL );
+		std::cout << "-- Running test 05: [1-row, n = " << n << " columns] matrix, filled with 1s" << std::endl;
+		input< NzType, void > input(I, mask);
+		if( launcher.exec( &grb_program, input, rc, true ) != SUCCESS ) {
+			std::cerr << "Launching test 04 FAILED\n";
+			return 255;
+		}
+		std::cout << std::endl << std::flush;
+	}
+
+	if( ! rc ) { // Building a [n rows, 1 column] matrix filled with 1s
+		Matrix< NzType > I( n, 1 );
+		std::vector< size_t > I_rows( n, 0 ), I_cols( n, 0 );
+		std::vector< NzType > I_vals( n, 1.f );
+		std::iota( I_rows.begin(), I_rows.end(), 0 );
+		buildMatrixUnique( I, I_rows.data(), I_cols.data(), I_vals.data(), I_vals.size(), SEQUENTIAL );
+		Matrix< void > mask( n, 1 );
+		buildMatrixUnique( mask, I_rows.data(), I_cols.data(), I_rows.size(), SEQUENTIAL );
+		std::cout << "-- Running test 06: [n = " << n << " rows, 1 column] matrix, filled with 1s" << std::endl;
+		input< NzType, void > input(I, mask);
+		if( launcher.exec( &grb_program, input, rc, true ) != SUCCESS ) {
+			std::cerr << "Launching test 06 FAILED\n";
+			return 255;
+		}
+		std::cout << std::endl << std::flush;
+	}
+
+	if( rc != SUCCESS ) {
+		std::cout << "Test FAILED (" << grb::toString( rc ) << ")" << std::endl;
+		return rc;
```

Comment:
the tests so far typically return an error code that corresponds to the failed subtest, which helps debugging without needing to involve gdb to see what failed where. I believe this test can effect that by having the helper subtest functions return an int error code that is unique to the `return` statement. Typically we use increments of `10` (rather than successive error numbers) to allow for the situation where a subtest may be inserted at some later stage